File: queues/ltcd_2444.cpp
 long long countSubarrays(vector<int>& nums, int minK, int maxK) 
    {
        long long last_out = -1;
        long long last_min = -1;
        long long last_max = -1;
        
        long long count    = 0;
        
        for (int i = 0; i < nums.size(); i++)
        {
            if (minK <= nums[i] && nums[i] <= maxK)
            {
                if (nums[i] == minK) last_min = i;
                if (nums[i] == maxK) last_max = i;
                count += max(0LL, min(last_min, last_max) - last_out);
            }
            else last_out = i;
        }

        return count;
    }

File: queues/ltcd_2444.py
    def countSubarrays(self, nums: List[int], minK: int, maxK: int) -> int:
        
        last_out = -1
        last_min = -1
        last_max = -1
        
        count = 0
        
        for i,n in enumerate(nums):
            
            if minK <= n <= maxK: 
                if n == minK : last_min = i
                if n == maxK : last_max = i
                count += max(min(last_min, last_max) - last_out, 0)
            else:
                last_out = i
            
        return count
